lms/djangoapps/branding/api.py

```diff
@@ -0,0 +1,110 @@
+    context["social_links"] = social_links()
+    context["about_links"] = about_edx_link(site_name)
+
+    return {"footer": context}
+
+
+def copy_right():
+    """ Returns the copy rights text
+    """
+    if settings.FEATURES.get('IS_EDX_DOMAIN', False):
+        data = _(
+            "(c) 2015 edX Inc. EdX, Open edX, and the edX and Open edX logos "
+            "are registered trademarks or trademarks of edX Inc."
+        )
+    else:
+        data = _(
+            "EdX, Open edX, and the edX and Open edX logos are registered trademarks or "
+            "trademarks of {link_start}edX Inc.{link_end}"
+        ).format(
+            link_start=u"<a href='https://www.edx.org/'>",
+            link_end=u"</a>"
+        )
+
+    return data
+
+
+def heading():
+    """ Returns the heading text
+    """
+    if settings.FEATURES.get('IS_EDX_DOMAIN', False):
+        data = _("EdX offers interactive online classes and MOOCs from the world's best universities, "
+                 "colleges and organizations. Online courses from MITx, HarvardX, BerkeleyX, UTx and "
+                 "many other universities can be taken here. Topics include biology, business, chemistry,"
+                 " computer science, economics, finance, electronics, engineering, food and nutrition,"
+                 " history, humanities, law, literature, math, medicine, music, philosophy, physics, science,"
+                 " statistics and more. EdX is a non-profit online initiative created by founding partners"
+                 " Harvard and MIT.")
+    else:
+        data = _("EdX offers interactive online classes and MOOCs from the world's best universities, "
+                 "colleges and organizations. Online courses from MITx, HarvardX, BerkeleyX, UTx and "
+                 "many other universities can be taken here. Topics include biology, business, chemistry,"
+                 " computer science, economics, finance, electronics, engineering, food and nutrition,"
+                 " history, humanities, law, literature, math, medicine, music, philosophy, physics, science,"
+                 " statistics and more. EdX is a non-profit online initiative created by founding partners"
+                 " Harvard and MIT.")
+    return data
+
+
+def social_links():
+    """ Returns the list of social link of footer
+    """
+    links = []
+    for social_name in settings.SOCIAL_MEDIA_FOOTER_NAMES:
+        links.append(
+            {
+                "provider": social_name,
+                "title": unicode(settings.SOCIAL_MEDIA_FOOTER_DISPLAY.get(social_name, {}).get("title", "")),
+                "url": settings.SOCIAL_MEDIA_FOOTER_URLS.get(social_name, "#")
+            }
+        )
+    return links
+
+
+def about_edx_link(site_name):
+    """ Returns the list of marketing links of footer
+    """
+
+    return dict(
+        [
+            (_(key), marketing_link(key))
```

Comment:
I'd recommend doing the same thing that the footer currently does, except in Python.  When we deprecate the old template, we can refactor it to pass this information in as template context.
```
return [
    {
        'title': _("About"),
        'url': marketing_link('about')
    },
    # ...
]
```
This has the added benefit of ensuring that the links appear in a consistent order.